### PR TITLE
Add sources to NU1507 warning

### LINKED ISSUE #11715: [DCR]: Print sources in NU1507
### NuGet Product(s) Affected

NuGet.exe, MSBuild.exe, dotnet.exe

### Current Behavior

As part of https://github.com/NuGet/Home/issues/11505, we emit a warning that only displays the number of sources in use.  

### Desired Behavior

We should just print the sources themselves in the warning because the sources can come from lots of places and at the moment its hard to determine what sources are in use and from where.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by jeffkl**
## Bug

Fixes: https://github.com/NuGet/Home/issues/11715

Regression? No Last working version:

## Description

Added the package source names to the NU1507 warning

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [x] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

File: src/NuGet.Core/NuGet.Commands/Strings.resx [hunk 1]
```diff
@@ -805,8 +805,8 @@ For more information, visit https://docs.nuget.org/docs/reference/command-line-r
     <comment>0 - The name of a package reference.</comment>
   </data>
   <data name="Warning_CentralPackageVersions_MultipleSourcesWithoutPackageSourceMapping" xml:space="preserve">
-    <value>There are {0} package sources defined in your configuration. When using central package management, please map your package sources with package source mapping (https://aka.ms/nuget-package-source-mapping) or specify a single package source.</value>
-    <comment>0 - The number of package sources.</comment>
+    <value>There are {0} package sources defined in your configuration. When using central package management, please map your package sources with package source mapping (https://aka.ms/nuget-package-source-mapping) or specify a single package source. The following sources are defined: {1}</value>
```
Comment (jeffkl) [inline src/NuGet.Core/NuGet.Commands/Strings.resx:808]: Team Review: Let's get clarification if its okay to use a comma-delimited list or some other pattern should be used

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs [hunk 2]
```diff
@@ -486,7 +486,7 @@ private async Task<bool> AreCentralVersionRequirementsSatisfiedAsync(RestoreRequ
             // Log a warning if there are more than one configured source and package source mapping is not enabled
             if (restoreRequest.Project.RestoreMetadata.Sources.Count > 1 && !restoreRequest.PackageSourceMapping.IsEnabled)
             {
-                await _logger.LogAsync(RestoreLogMessage.CreateWarning(NuGetLogCode.NU1507, string.Format(CultureInfo.CurrentCulture, Strings.Warning_CentralPackageVersions_MultipleSourcesWithoutPackageSourceMapping, restoreRequest.Project.RestoreMetadata.Sources.Count)));
+                await _logger.LogAsync(RestoreLogMessage.CreateWarning(NuGetLogCode.NU1507, string.Format(CultureInfo.CurrentCulture, Strings.Warning_CentralPackageVersions_MultipleSourcesWithoutPackageSourceMapping, restoreRequest.Project.RestoreMetadata.Sources.Count, string.Join(", ", restoreRequest.Project.RestoreMetadata.Sources.Select(i => i.Name)))));
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs:489]: Does the url matter more or the name?

File: src/NuGet.Core/NuGet.Commands/Strings.resx (hunk 1 above)
Comment (nkolev92) [reply src/NuGet.Core/NuGet.Commands/Strings.resx:808]: I've used comma delimiters for these purposes before. Seems reasonable to me.

**Review by nkolev92 — COMMENTED** [review]
One clarification
PackageSource.Name or PackageSource.Source, that is name vs url.

File: src/NuGet.Core/NuGet.Commands/Strings.resx [hunk 3]
```diff
@@ -805,8 +805,8 @@ For more information, visit https://docs.nuget.org/docs/reference/command-line-r
...
+    <comment>0 - The number of package sources, 1 - The URL of the package sources.</comment>
```
Comment (dominoFire) [inline src/NuGet.Core/NuGet.Commands/Strings.resx]: I would add a note saying that URLs will be comma separated. Let's wait for loc review.

```suggestion
    <comment>0 - The number of package sources, 1 - The URL of the package sources, comma separated.</comment>
```

Credits to @donnie-msft for TFS example.

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs (hunk 2 above)
Comment (jeffkl) [reply src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs:489]: I'm not sure, I originally was calling the `.ToString()` method on `PackageSource` but it returns the format:

https://github.com/NuGet/NuGet.Client/blob/336320cc799d5a4e36da439148523e5ab9e4ff1e/src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSource.cs#L186-L189

Which looks funky when the name and source are the same value.  Can I update the `ToString()` method to something like this instead:

```c#
public override string ToString() 
{
    if (Name == Source)
    {
        return Source;
    }
    else
    {
        return Name + " [" + Source + "]";
    }
} 
```

Or would that be considered a breaking change?  I think its useful to have both the name and the source but not if they are the same value.  I guess I could just add that logic here instead.  Thoughts?

Comment (nkolev92) [reply src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs:489]: No clue what it breaks. 

We probably use the sources in a bunch of other error messages, we should probably match that. https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu1102

It uses the name. No one has really complained, maybe best we do not overthink it :) 

I'll approve. If you decide to go a different direction, happy to approve that as well.

**Review by nkolev92 — APPROVED** [review]

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs (hunk 2 above)
Comment (jeffkl) [reply src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs:489]: I think I'll leave name in there for now, we can adjust it later if we get feedback.  Just need to get CI passing

**Pull request merged**